extensions/bitwarden/src/search.tsx

```diff
@@ -0,0 +1,191 @@
+import treeify from "treeify";
+import { filterNullishPropertiesFromObject, codeBlock, titleCase } from "./utils";
+import { useBitwarden as useSessionToken } from "./hooks";
+import { TroubleshootingGuide, UnlockForm } from "./components";
+import { existsSync } from "fs";
+import { BitwardenApi } from "./api";
+import { dirname } from "path";
+
+const { cliPath, clientId, clientSecret, fetchFavicons } = getPreferenceValues();
+process.env.PATH = dirname(cliPath)
+const bitwardenApi = new BitwardenApi(clientId, clientSecret);
+
+export default function ListCommand(): JSX.Element {
+  if (!existsSync(cliPath)) {
+    return <TroubleshootingGuide />;
+  }
+
+  const [sessionToken, setSessionToken] = useSessionToken(bitwardenApi);
+
+  if (sessionToken === null) return <UnlockForm setSessionToken={setSessionToken} bitwardenApi={bitwardenApi} />;
+
+  return <ItemList bitwardenApi={bitwardenApi} sessionToken={sessionToken}/>;
+}
+
+function ItemList(props: { bitwardenApi: BitwardenApi; sessionToken: string | undefined }) {
+  const { bitwardenApi, sessionToken } = props;
+  const [state, setState] = useState<{ folders: Folder[]; items: Item[] }>();
+
+  const folderMap: Record<string, Folder> = {};
+  for (const folder of state?.folders || []) {
+    if (folder.id) folderMap[folder.id] = folder;
+  }
+
+  async function loadItems(sessionToken: string) {
+    try {
+      const [items, folders] = await Promise.all([
+        bitwardenApi.listItems<Item>("items", sessionToken),
+        bitwardenApi.listItems<Folder>("folders", sessionToken),
+      ]);
+
+      setState({ items, folders });
+    } catch (error) {
+      showToast(ToastStyle.Failure, "Failed to search vault", "Vault is locked");
+    }
+  }
+
+  useEffect(() => {
+    if (sessionToken) loadItems(sessionToken);
+  }, [sessionToken]);
+
+  return (
+    <List isLoading={typeof state === "undefined"}>
+      {state?.items.map((item) => (
+        <ItemListItem
+          key={item.id}
+          item={item}
+          folder={item.folderId ? folderMap[item.folderId] : undefined}
+          refreshItems={async () => {
+            if (sessionToken) {
+              const toast = await showToast(ToastStyle.Animated, "Syncing Items...");
+              await bitwardenApi.syncItems(sessionToken)
+              await loadItems(sessionToken);
+              await toast.hide();
+            }
+          }}
+        />
+      ))}
+    </List>
+  );
+}
+
+function ItemListItem(props: { item: Item; folder: Folder | undefined; refreshItems?: () => void }) {
+  const { item, folder, refreshItems } = props;
+  const { type, name, notes, identity, login, secureNote, fields, passwordHistory, card } = item;
+  const accessoryIcons = [];
+  if (folder) accessoryIcons.push(`📂 ${folder.name}`);
+  if (item.favorite) accessoryIcons.push(`⭐️`);
+
+  const fieldMap = Object.fromEntries(fields?.map((field) => [field.name, field.value]) || []);
+  const uriMap = Object.fromEntries(
```

Comment:
```suggestion
  const fields = Object.fromEntries(fields?.map((field) => [field.name, field.value]) || []);
  const URIs = Object.fromEntries(
```

Reply:
This would not work since the fields const is already defined. Why do you want to remove the map prefix ?
I feel like the code is more readable this way.

Reply:
It's almost never a good idea for a variable's name to include the type of the object stored in the variable, even when using `const`. It just makes the code more verbose. Since `fields` happens to be taken already (I didn't notice), we can leave it as-is here.